src/components/UI/ModalForm.tsx

```diff
@@ -0,0 +1,215 @@
+import { PropsWithoutRef, ReactNode, useEffect, useState } from 'react';
+import { User } from '@/api/users/user.interface';
+import { useAppTranslation } from '@/lib/i18n';
+import Icons from '@/icons';
+import { UserCard } from '../users/UserCard';
+import UserSelector from '../users/UserSelector';
+import Input from './Input';
+import Button from './Button';
+import styles from './ModalForm.module.scss';
+
+interface DataModalType {
+  date: Date;
+  user: User;
+  string: string;
+  stringList: string[];
+}
+
+// These options are already of type []
+type OptionsFieldsRequired = 'stringList';
+// These options must NOT be of type []
+type OptionsFieldsPossible = 'string';
+
+interface DataModalEntryBase<T extends keyof DataModalType> {
+  type: T;
+  defaultValue?: DataModalType[T];
+  /** When using options, you can choose to add label. The label must be provided as first element of the tuple */
+  options: DataModalType[T] | [DataModalType[T], DataModalType[T]];
+  label: ReactNode;
+  required?: boolean;
+}
+
+type DataModalEntry<T extends keyof DataModalType> = T extends OptionsFieldsRequired
+  ? DataModalEntryBase<T>
+  : T extends OptionsFieldsPossible
+    ? Omit<DataModalEntryBase<T>, 'options'> & { options?: DataModalEntryBase<T>['options'][] }
+    : Omit<DataModalEntryBase<T>, 'options'>;
+
+type DataModalKeys = { [key: string]: keyof DataModalType };
+
+export type DataModalSchema<T extends DataModalKeys> = {
+  [S in keyof T]: DataModalEntry<T[S]>;
```

Comment:
Idem ici, je comprends pas à quoi sert `DataModalEntry`, puisque `T[S]` vaut toujours `Date | User | string | string[]`, donc de ce que je comprends, `DataModalEntry<T[S]>` vaut nécessairement `Omit<DataModalEntryBase<T>, 'options'>`

Reply:
_Hésite pas à ouvrir in ide pour voir ce que ça donne_